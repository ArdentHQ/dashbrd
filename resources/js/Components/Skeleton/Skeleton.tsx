import SkeletonReact from "react-loading-skeleton";
import { twMerge } from "tailwind-merge";

interface SkeletonProperties {
    width?: string | number;
    height?: string | number;
    isCircle?: boolean;
    animated?: boolean;
    className?: string;
}

export const Skeleton = ({
    width,
    height,
    animated = true,
    isCircle = false,
    className,
}: SkeletonProperties): JSX.Element => (
    <SkeletonReact
        enableAnimation={animated}
        containerTestId="Skeleton"
        circle={isCircle}
        style={{ width, height }}
        className={twMerge("z-0 dark:bg-theme-dark-800", !isCircle && "rounded-lg", className)}
        duration={1.3}
        containerClassName="flex w-auto max-w-full items-center leading-none h-full"
    />
);
