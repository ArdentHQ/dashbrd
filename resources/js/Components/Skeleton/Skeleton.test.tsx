import React from "react";
import { Skeleton } from "./Skeleton";
import { render, screen } from "@/Tests/testing-library";

describe("Skeleton", () => {
    it("should render", () => {
        const { asFragment } = render(<Skeleton />);

        expect(screen.getByTestId("Skeleton")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render circle", () => {
        const { asFragment } = render(<Skeleton isCircle />);

        expect(screen.getByTestId("Skeleton")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});
