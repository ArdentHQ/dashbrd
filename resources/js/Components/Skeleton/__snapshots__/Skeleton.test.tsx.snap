// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > should render 1`] = `
<DocumentFragment>
  <span
    aria-busy="true"
    aria-live="polite"
    class="flex w-auto max-w-full items-center leading-none h-full"
    data-testid="Skeleton"
  >
    <span
      class="react-loading-skeleton z-0 dark:bg-theme-dark-800 rounded-lg"
      style="--animation-duration: 1.3s;"
    >
      ‌
    </span>
    <br />
  </span>
</DocumentFragment>
`;

exports[`Skeleton > should render circle 1`] = `
<DocumentFragment>
  <span
    aria-busy="true"
    aria-live="polite"
    class="flex w-auto max-w-full items-center leading-none h-full"
    data-testid="Skeleton"
  >
    <span
      class="react-loading-skeleton z-0 dark:bg-theme-dark-800"
      style="--animation-duration: 1.3s; border-radius: 50%;"
    >
      ‌
    </span>
    <br />
  </span>
</DocumentFragment>
`;
